private/astarte_linked_list.h

```diff
@@ -0,0 +1,137 @@
+/*
+ * (C) Copyright 2023, SECO Mind Srl
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR Apache-2.0
+ */
+
+/**
+ * @file astarte_linked_list.h
+ * @brief Utility module containing a doubly linked list implementation.
+ *
+ * @details This library does not perform deep copies when storing values.
+ * The user should ensure that any value is freed correctly when appropriate.
+ */
+
+#ifndef _ASTARTE_LINKED_LIST_H_
+#define _ASTARTE_LINKED_LIST_H_
+
+#include <stdbool.h>
+#include <stdlib.h>
+
+#include "astarte.h"
+
+typedef struct
+{
+    struct astarte_linked_list_node *head;
+    struct astarte_linked_list_node *tail;
+} astarte_linked_list_handle_t;
+
+typedef struct
+{
+    astarte_linked_list_handle_t *handle;
+    struct astarte_linked_list_node *node;
+} astarte_linked_list_iterator_t;
+
+/**
+ * @brief Initializes a new empty linked list
+ *
+ * @return A handle to the newly initialized list
+ */
+astarte_linked_list_handle_t astarte_linked_list_init(void);
+
+/**
+ * @brief Checks if the linked list is empty
+ *
+ * @param[in] handle Linked list handle
+ * @return true if the list is empty, false otherwise
+ */
+bool astarte_linked_list_is_empty(astarte_linked_list_handle_t *handle);
+
+/**
+ * @brief Append an item to the end of a linked list
```

Comment:
Use 3rd person (descriptive) not 2nd person (prescriptive).


```suggestion
 * @brief Appends an item to the end of a linked list
```